lib/deb/s3/release.rb

```diff
@@ -102,9 +102,12 @@ def write_to_s3
 
     # sign the file, if necessary
     if Deb::S3::Utils.signing_key
+      local_file = release_tmp.path+".asc"
       key_param = Deb::S3::Utils.signing_key != "" ? "--default-key=#{Deb::S3::Utils.signing_key}" : ""
-      if system("gpg -a #{key_param} --digest-algo SHA256 #{Deb::S3::Utils.gpg_options} -s --clearsign #{release_tmp.path}")
-        local_file = release_tmp.path+".asc"
+      tmp_gpg_cmd = Deb::S3::Utils.gpg_cmd != "" ? Deb::S3::Utils.gpg_cmd + " --in #{release_tmp.path} --out #{local_file}" : nil
```

Comment:
I'd rather not put the `--in`/`--out` logic in deb-s3. Instead, we can do blind string replacements on `INPUT` and `OUTPUT`

Reply:
I don't care either way, but I think this means the script have advance knowledge that the string contains INPUT and OUTPUT, which would highly couple it to that input string. 

I think whichever approach we take, we're going to have to create a heptio fork and work from there. This code will only matter to us and perhaps a very small subset of users to be merged upstream.